### PR TITLE
feat(cursor): Add Cursor IDE module

### PR DESCRIPTION
**Pull request by matifali**
This PR introduces the Cursor IDE module, allowing users to add
a one-click button for launching Cursor IDE into their workspaces.

- Adds a comprehensive README for setting up the module.
- Includes example configurations for typical usage scenarios.
- Provides automated tests to ensure module reliability.

commit generetaed by [aicommit](https://github.com/coder/aicommit)

### CODE REVIEW

**Review by matifali — COMMENTED** [review]
to test please add the follwing snippet to your template

```tf
module "cursor" {
  source    = "git::https://github.com/coder/modules.git//cursor?ref=cursor"
  agent_id = coder_agent.example.id 
}
```

**Comment by stirby** [comment]
Using this in my workspace, on the first attempt my healthy workspace returned a 404, offered to retry. Second attempt was fine.

Icon still shows VSCode.

**Comment by stirby** [comment]
Screenshot 2024-09-19 at 2 05 44 PM

File: cursor/main.tf [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  default     = null
+}
+
+data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
+
+resource "coder_app" "cursor" {
+  agent_id     = var.agent_id
+  external     = true
+  icon         = "/icon/code.svg"
```
Comment (stirby) [inline cursor/main.tf:41]: Code icon

File: cursor/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+description: Add a one-click button to launch Cursor IDE
+icon: ../.icons/cursor.svg
+maintainer_github: coder
+verified: true
+tags: [ide, cursor, helper]
+---
+
+# Cursor IDE
+
+Add a button to open any workspace with a single click in Cursor IDE.
```
Comment (stirby) [inline cursor/README.md:12]: "Requires the local Coder VSCode extension. You can [import this extension into Cursor](https://docs.cursor.com/get-started/migrate-from-vscode)."

**Review by stirby — CHANGES_REQUESTED** [review]

File: cursor/README.md (hunk 2 above)
Comment (matifali) [reply cursor/README.md:12]: For me it auto prompts to install the extension if not installed already (Similar to VS Code).
image

File: cursor/main.tf (hunk 1 above)
Comment (matifali) [reply cursor/main.tf]: This also needs adding the icon to `coder/coder`

Comment (matifali) [reply cursor/main.tf]: https://github.com/coder/coder/pull/14743

**Comment by stirby** [comment]
Works fine now on both MacOS and Windows, maybe it was an issue with my local cursor.

**Review by stirby — APPROVED** [review]
❤️

**Pull request merged**